src/internal.h

```diff
@@ -230,7 +230,11 @@ DISPATCH_EXPORT DISPATCH_NOTHROW void dispatch_atfork_child(void);
 #include <search.h>
 #if USE_POSIX_SEM
 #include <semaphore.h>
+#elif USE_FUTEX_SEM
```

Comment:
since USE_FUTEX will probably be used for more than semaphores, I would test for USE_FUTEX first, as it's "more" generic.